Subject: Patch MSC_VER for MSVC 17.10.2 (_MSC_VER is 1940)

diff --git win32/setup.mak win32/setup.mak
--- win32/setup.mak
+++ win32/setup.mak
@@ -157,7 +157,10 @@ echo TEENY = RUBY_VERSION_TEENY
 echo RUBY_DEVEL = yes
 #endif
 set /a MSC_VER = _MSC_VER
-#if _MSC_VER > 1900
+#if _MSC_VER >= 1920
+set /a MSC_VER_LOWER = 1920
+set /a MSC_VER_UPPER = 1959
+#elif _MSC_VER >= 1900
 set /a MSC_VER_LOWER = MSC_VER/10*10+0
 set /a MSC_VER_UPPER = MSC_VER/10*10+9
 #endif
