diff --git load.c load.c
--- load.c
+++ load.c
@@ -1334,11 +1334,14 @@ register_init_ext(st_data_t *key, st_data_t *value, st_data_t init, int existing
 void
 ruby_init_ext(const char *name, void (*init)(void))
 {
+    rb_warn("ruby_init_ext for %s", name);
     rb_vm_t *vm = GET_VM();
     st_table *inits_table = vm->static_ext_inits;
 
-    if (feature_provided(vm, name, 0))
+    if (feature_provided(vm, name, 0)) {
+        rb_warn("feature_provided for %s", name);
         return;
+    }
     st_update(inits_table, (st_data_t)name, register_init_ext, (st_data_t)init);
 }
 #endif
