diff --git a/template/Makefile.in b/template/Makefile.in
--- template/Makefile.in
+++ template/Makefile.in
@@ -333,7 +333,18 @@ $(LIBRUBY_A):
 		@$(RM) $@
 		@-[ -z "$(EXTSTATIC)" ] || $(PRE_LIBRUBY_UPDATE)
 		$(ECHO) linking static-library $@
-		$(Q) $(AR) $(ARFLAGS) $@ $(LIBRUBY_A_OBJS) $(INITOBJS)
+		$(ECHO) LIBRUBY_A_OBJS=$(LIBRUBY_A_OBJS)
+		$(ECHO) INITOBJS=$(INITOBJS)
+		$(ECHO) EXTOBJS=$(EXTOBJS)
+		$(ECHO) ENCOBJS=$(ENCOBJS)
+		$(ECHO) ext/extinit.$(OBJEXT) enc/encinit.$(OBJEXT)
+		$(Q) if [ -z "$(EXTSTATIC)" ]; then \
+		  $(ECHO0) CMD=$(AR) $(ARFLAGS) $@ $(LIBRUBY_A_OBJS) $(INITOBJS); \
+		  $(AR) $(ARFLAGS) $@ $(LIBRUBY_A_OBJS) $(INITOBJS); \
+		else \
+		  $(ECHO0) CMD=$(AR) $(ARFLAGS) $@ $(LIBRUBY_A_OBJS) $(INITOBJS); \
+		  $(AR) $(ARFLAGS) $@ $(LIBRUBY_A_OBJS) $(INITOBJS); \
+		fi
 		$(Q) if [ 'no' != '$(YJIT_SUPPORT)' ]; then \
 		  set -eu$(V0:1=x) && \
 		  $(ECHO0) 'merging $(YJIT_LIBS) into $@' && \
@@ -346,7 +357,18 @@ $(LIBRUBY_A):
 		fi
 		@-$(RANLIB) $@ 2> /dev/null || true
 
-verify-static-library: $(LIBRUBY_A)
+final-libruby_a:
+		@-[ -z "$(EXTSTATIC)" ]
+		$(ECHO) Replacing INCOBJS with EXTOBJS and ENCOBS in static-library $(LIBRUBY_A)
+		$(ECHO) LIBRUBY_A_OBJS=$(LIBRUBY_A_OBJS)
+		$(ECHO) INITOBJS=$(INITOBJS)
+		$(ECHO) EXTOBJS=$(EXTOBJS)
+		$(ECHO) ENCOBJS=$(ENCOBJS)
+		$(AR) dD $(LIBRUBY_A) $(INITOBJS)
+		$(AR) rD $(LIBRUBY_A) $(EXTOBJS) $(ENCOBJS)
+		@-$(RANLIB) $(LIBRUBY_A) 2> /dev/null || true
+
+verify-static-library:
 		$(ECHO) verifying static-library $@
 		@$(PURIFY) $(CC) $(EXE_LDFLAGS) $(XLDFLAGS) $(MAINOBJ) $(LIBRUBY_A) $(MAINLIBS) $(EXTLIBS) $(LIBS) $(OUTFLAG)conftest$(EXEEXT)
 		@$(RMALL) conftest$(EXEEXT) conftest.c conftest.dSYM
