diff --git a/lib/mkmf.rb b/lib/mkmf.rb
--- lib/mkmf.rb
+++ lib/mkmf.rb
@@ -2792,7 +2792,21 @@ MESSAGE
   ##
   # A C main function which does no work
 
-  MAIN_DOES_NOTHING = config_string('MAIN_DOES_NOTHING') || "int main(int argc, char **argv)\n{\n  return !!argv[argc];\n}"
+  MAIN_DOES_NOTHING = config_string('MAIN_DOES_NOTHING') || <<-SRC
+int main(int argc, char **argv)
+{
+  return !!argv[argc];
+}
+
+int rb_hasFile(const char *t_filename) {
+  return 0;
+}
+
+int rb_require_embedded(const char *t_filename) {
+  return 0;
+}
+SRC
+
   UNIVERSAL_INTS = config_string('UNIVERSAL_INTS') {|s| Shellwords.shellwords(s)} ||
     %w[int short long long\ long]
 
diff --git a/lib/rubygems/util.rb b/lib/rubygems/util.rb
--- lib/rubygems/util.rb
+++ lib/rubygems/util.rb
@@ -98,7 +98,9 @@ module Gem::Util
 
   def self.glob_files_in_dir(glob, base_path)
     if RUBY_VERSION >= "2.5"
-      Dir.glob(glob, base: base_path).map! {|f| File.expand_path(f, base_path) }
+      Dir.glob(File.expand_path(glob, base_path))
+      #Below fails on windows. Throws invalid argument"
+      #Dir.glob(glob, base: base_path).map! {|f| File.expand_path(f, base_path) }
     else
       Dir.glob(File.expand_path(glob, base_path))
     end
diff --git a/load.c b/load.c
--- load.c
+++ load.c
@@ -17,6 +17,9 @@
 #include "ruby/encoding.h"
 #include "ruby/util.h"
 
+int rb_hasFile(const char *t_filename);
+int rb_require_embedded(const char *t_filename);
+
 static VALUE ruby_dln_librefs;
 
 #define IS_RBEXT(e) (strcmp((e), ".rb") == 0)
@@ -1235,7 +1238,11 @@ rb_require_string(VALUE fname)
 VALUE
 rb_require(const char *fname)
 {
+  if ( rb_hasFile(fname) ) {
+    return rb_require_embedded(fname);
+  } else {
     return rb_require_string(rb_str_new_cstr(fname));
+  }
 }
 
 static int
diff --git a/main.c b/main.c
--- main.c
+++ main.c
@@ -47,3 +47,11 @@ main(int argc, char **argv)
 	return ruby_run_node(ruby_options(argc, argv));
     }
 }
+
+int rb_hasFile(const char *t_filename) {
+  return 0;
+}
+
+int rb_require_embedded(const char *t_filename) {
+  return 0;
+}
